src/main/java/org/folio/rest/impl/DataExportImpl.java

```diff
@@ -87,11 +91,23 @@ public void postDataExportExport(ExportRequest entity, Map<String, String> okapi
                   .onSuccess(jobExecution ->
                     jobExecutionService.update(jobExecution.withJobProfileId(jobProfile.getId()), tenantId)
                       .onSuccess(updatedJobExecution -> {
-                        inputDataManager.init(JsonObject.mapFrom(entity), JsonObject.mapFrom(requestFileDefinition), JsonObject.mapFrom(mappingProfile), JsonObject.mapFrom(updatedJobExecution), okapiHeaders);
                         succeededFuture()
                           .map(PostDataExportExportResponse.respond204())
                           .map(Response.class::cast)
                           .onComplete(asyncResultHandler);
+                        AtomicLong periodicId = new AtomicLong();
+                        AtomicReference<FileDefinition> fileDefinitionAtomicReference = new AtomicReference<>(requestFileDefinition);
+                        periodicId.set(vertxContext.owner().setPeriodic(1000, handler -> {
+                          if (nonNull(fileDefinitionAtomicReference.get().getSourcePath()) ||
```

Comment:
Please, add conditions to separate method the name of which represents what is checked here.